diff --git a/BUILD b/BUILD
index 29af922edc..1ce64a417e 100644
--- a/BUILD
+++ b/BUILD
@@ -1831,6 +1831,7 @@ grpc_cc_library(
         "//src/core:tsi/transport_security_grpc.cc",
     ],
     hdrs = [
+        "include/grpc/slice_buffer.h",
         "//src/core:tsi/transport_security.h",
         "//src/core:tsi/transport_security_grpc.h",
         "//src/core:tsi/transport_security_interface.h",
diff --git a/bazel/grpc_deps.bzl b/bazel/grpc_deps.bzl
index 4740bbd98e..3ee8c15eeb 100644
--- a/bazel/grpc_deps.bzl
+++ b/bazel/grpc_deps.bzl
@@ -263,17 +263,11 @@ def grpc_deps():
     if "com_google_protobuf" not in native.existing_rules():
         http_archive(
             name = "com_google_protobuf",
-            sha256 = "70f480fe9cb0c6829dbf6be3c388103313aacb65de667b86d981bbc9eaedb905",
-            strip_prefix = "protobuf-7f94235e552599141950d7a4a3eaf93bc87d1b22",
+            sha256 = "3ae7a8f2181be28e5d694617c2b85c6561ba6a16bfcdc9db8e3a95077cab8815",
+            strip_prefix = "protobuf-25.3",
             urls = [
-                # https://github.com/protocolbuffers/protobuf/commits/v25.0
-                "https://storage.googleapis.com/grpc-bazel-mirror/github.com/protocolbuffers/protobuf/archive/7f94235e552599141950d7a4a3eaf93bc87d1b22.tar.gz",
-                "https://github.com/protocolbuffers/protobuf/archive/7f94235e552599141950d7a4a3eaf93bc87d1b22.tar.gz",
+                "https://github.com/protocolbuffers/protobuf/releases/download/v25.3/protobuf-25.3.zip",
             ],
-            patches = [
-                "@com_github_grpc_grpc//third_party:protobuf.patch",
-            ],
-            patch_args = ["-p1"],
         )
 
     if "com_google_googletest" not in native.existing_rules():
diff --git a/examples/cpp/helloworld/BUILD b/examples/cpp/helloworld/BUILD
index 463dd6f495..94e2f02ca5 100644
--- a/examples/cpp/helloworld/BUILD
+++ b/examples/cpp/helloworld/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+package(default_visibility = ["//visibility:public"])
+
 licenses(["notice"])
 
 cc_binary(
