diff --git a/bin/kmstool-enclave-cli/main.c b/bin/kmstool-enclave-cli/main.c
index f810459..9a7ffd5 100644
--- a/bin/kmstool-enclave-cli/main.c
+++ b/bin/kmstool-enclave-cli/main.c
@@ -170,9 +170,9 @@ static void s_parse_options(int argc, char **argv, const char *subcommand, struc
                 break;
             case 'h':
                 if (strncmp(subcommand, DECRYPT_CMD, MAX_SUB_COMMAND_LENGTH) == 0)
-                    s_usage_decrypt(1);
+                    s_usage_decrypt(0);
                 else if (strncmp(subcommand, GENKEY_CMD, MAX_SUB_COMMAND_LENGTH) == 0)
-                    s_usage_genkey(1);
+                    s_usage_genkey(0);
                 break;
             default:
                 if (strncmp(subcommand, DECRYPT_CMD, MAX_SUB_COMMAND_LENGTH) == 0) { 
@@ -268,6 +268,13 @@ static void init_kms_client(struct app_ctx *app_ctx, struct aws_credentials **cr
     struct aws_nitro_enclaves_kms_client_configuration configuration = {
         .allocator = app_ctx->allocator, .endpoint = &endpoint, .domain = AWS_SOCKET_VSOCK, .region = app_ctx->region};
 
+    if (app_ctx->proxy_port == 0) {
+        fprintf(stderr, "app_ctx->proxy_port is %d\n ", app_ctx->proxy_port);
+        configuration.endpoint = NULL;
+    }
+
+    fprintf(stderr, "app_ctx->proxy_port is %d\n ", app_ctx->proxy_port);
+
     /* Sets the AWS credentials and creates a KMS client with them. */
     struct aws_credentials *new_credentials = aws_credentials_new(
         app_ctx->allocator,
