diff --git a/api/include/opentelemetry/nostd/type_traits.h b/api/include/opentelemetry/nostd/type_traits.h
index 3a64b5db1..3517bcc52 100644
--- a/api/include/opentelemetry/nostd/type_traits.h
+++ b/api/include/opentelemetry/nostd/type_traits.h
@@ -3,6 +3,9 @@
 
 #pragma once
 
+#include "opentelemetry/config.h"
+#include "opentelemetry/version.h"
+
 #if defined(OPENTELEMETRY_STL_VERSION)
 #  if OPENTELEMETRY_STL_VERSION >= 2011
 #    include "opentelemetry/std/type_traits.h"
@@ -14,9 +17,7 @@
 #  include <array>
 #  include <type_traits>
 
-#  include "opentelemetry/config.h"
 #  include "opentelemetry/nostd/detail/void.h"  // IWYU pragma: export
-#  include "opentelemetry/version.h"
 #endif
 
 OPENTELEMETRY_BEGIN_NAMESPACE