diff --git a/sdk/include/opentelemetry/sdk/metrics/data/circular_buffer.h b/sdk/include/opentelemetry/sdk/metrics/data/circular_buffer.h
index 1608a3a0..26f272f8 100644
--- a/sdk/include/opentelemetry/sdk/metrics/data/circular_buffer.h
+++ b/sdk/include/opentelemetry/sdk/metrics/data/circular_buffer.h
@@ -5,6 +5,7 @@

 #include "opentelemetry/nostd/variant.h"

+#include <cstdint>
 #include <limits>
 #include <vector>

 