diff --git a/BUILD.boost b/BUILD.boost
index 93e03f9..0fe10e7 100644
--- a/BUILD.boost
+++ b/BUILD.boost
@@ -438,7 +438,7 @@ cc_library(
     visibility = ["//visibility:public"],
     deps = [
         ":asio",
-        "@openssl//:ssl",
+        "@boringssl//:ssl",
     ],
 )

diff --git a/boost/boost.bzl b/boost/boost.bzl
index ff1630f..c483268 100644
--- a/boost/boost.bzl
+++ b/boost/boost.bzl
@@ -219,10 +219,10 @@ def boost_deps():
         ],
     )

-    maybe(
-        http_archive,
-        name = "openssl",
-        sha256 = "6f640262999cd1fb33cf705922e453e835d2d20f3f06fe0d77f6426c19257308",
-        strip_prefix = "boringssl-fc44652a42b396e1645d5e72aba053349992136a",
-        url = "https://github.com/google/boringssl/archive/fc44652a42b396e1645d5e72aba053349992136a.tar.gz",
-    )
+#     maybe(
+#         http_archive,
+#         name = "openssl",
+#         sha256 = "6f640262999cd1fb33cf705922e453e835d2d20f3f06fe0d77f6426c19257308",
+#         strip_prefix = "boringssl-fc44652a42b396e1645d5e72aba053349992136a",
+#         url = "https://github.com/google/boringssl/archive/fc44652a42b396e1645d5e72aba053349992136a.tar.gz",
+#     )
